toolchain/lower/file_context.cpp

```diff
@@ -311,14 +321,22 @@ auto FileContext::BuildFunctionDefinition(SemIR::FunctionId function_id)
     return;
   }
 
-  llvm::Function* llvm_function = GetFunction(function_id);
-  if (!llvm_function) {
-    // We chose not to lower this function at all, for example because it's a
-    // generic function.
-    return;
+  llvm::Function* llvm_function;
+  if (!specific_id.has_value()) {
+    llvm_function = GetFunction(function_id);
+    if (!llvm_function) {
+      // We chose not to lower this function at all, for example because it's a
+      // generic function.
+      return;
+    }
+  } else {
+    llvm_function = specific_functions_[specific_id.index];
   }
```

Comment:
This might be more legible inverted - "if (specific_id.has_value()) { llvm_function = specific_functions_[specific_id.index]; } else { ... }" (then the test/use of specific_id are close together)